packages/components/src/gridicon/index.tsx

```diff
@@ -17,11 +17,36 @@ interface Props {
 
 type AllProps = Assign< React.SVGProps< SVGSVGElement >, Props >;
 
+/**
+ * Fetches the SVG file and creates a proxy URL for it.
+ * @param url the original cross-origin URL
+ * @returns the proxied URL.
+ */
+function useProxiedURL( url: string ) {
+	const [ urlProxy, setUrlProxy ] = React.useState( url );
+	React.useEffect( () => {
+		// Only bother with cross-origin absolute URLs.
+		if ( url.startsWith( 'https://' ) && window.location.origin !== new URL( url ).origin ) {
```

Comment:
This code has two downsides:
- the URL used in the `use` tag will be the original cross-origin one while the fetch is in progress
- the `new URL` constructor can throw an exception if the `url` is not OK. Being relative is enough to fail.